src/sql-parser/src/parser.rs

```diff
@@ -506,6 +506,17 @@ impl<'a> Parser<'a> {
 
     /// Parse an expression prefix
     fn parse_prefix(&mut self) -> Result<Expr<Raw>, ParserError> {
+        // Moving the timestamp out of the maybe_parse below so that
+        // the errors are not swallowed.
+        if let Some(Token::Keyword(TIMESTAMP)) | Some(Token::Keyword(TIMESTAMPTZ)) =
+            self.peek_token()
+        {
+            let data_type = self.parse_data_type()?;
+            return Ok(Expr::Cast {
+                expr: Box::new(Expr::Value(Value::String(self.parse_literal_string()?))),
+                data_type,
+            });
+        }
```

Comment:
The parsing errors were getting swallowed, so checking the timestamp ones before the the `maybe` block below.

Reply:
Moved this after chatting offline with @sploiselle